atom/common/platform_util_win.cc

```diff
@@ -169,33 +170,298 @@ bool OpenExternal(const GURL& url) {
+    /* [in] */ DWORD dwFlags,
+    /* [in] */ __RPC__in_opt IShellItem *psiItem,
+    /* [string][in] */ __RPC__in_string LPCWSTR pszNewName,
+    /* [in] */ HRESULT hrRename,
+    /* [in] */ __RPC__in_opt IShellItem *psiNewlyCreated);
+    
+    HRESULT STDMETHODCALLTYPE PreMoveItem( 
+    /* [in] */ DWORD dwFlags,
+    /* [in] */ __RPC__in_opt IShellItem *psiItem,
+    /* [in] */ __RPC__in_opt IShellItem *psiDestinationFolder,
+    /* [string][unique][in] */ __RPC__in_opt_string LPCWSTR pszNewName);
+    
+    HRESULT STDMETHODCALLTYPE PostMoveItem( 
+    /* [in] */ DWORD dwFlags,
+    /* [in] */ __RPC__in_opt IShellItem *psiItem,
+    /* [in] */ __RPC__in_opt IShellItem *psiDestinationFolder,
+    /* [string][unique][in] */ __RPC__in_opt_string LPCWSTR pszNewName,
+    /* [in] */ HRESULT hrMove,
+    /* [in] */ __RPC__in_opt IShellItem *psiNewlyCreated);
+    
+    HRESULT STDMETHODCALLTYPE PreCopyItem( 
+    /* [in] */ DWORD dwFlags,
+    /* [in] */ __RPC__in_opt IShellItem *psiItem,
+    /* [in] */ __RPC__in_opt IShellItem *psiDestinationFolder,
+    /* [string][unique][in] */ __RPC__in_opt_string LPCWSTR pszNewName);
+    
+    HRESULT STDMETHODCALLTYPE PostCopyItem( 
+    /* [in] */ DWORD dwFlags,
+    /* [in] */ __RPC__in_opt IShellItem *psiItem,
+    /* [in] */ __RPC__in_opt IShellItem *psiDestinationFolder,
+    /* [string][unique][in] */ __RPC__in_opt_string LPCWSTR pszNewName,
+    /* [in] */ HRESULT hrCopy,
+    /* [in] */ __RPC__in_opt IShellItem *psiNewlyCreated);
+    
+    HRESULT STDMETHODCALLTYPE PreDeleteItem( 
+    /* [in] */ DWORD dwFlags,
+    /* [in] */ __RPC__in_opt IShellItem *psiItem);
+    
+    HRESULT STDMETHODCALLTYPE PostDeleteItem( 
+    /* [in] */ DWORD dwFlags,
+    /* [in] */ __RPC__in_opt IShellItem *psiItem,
+    /* [in] */ HRESULT hrDelete,
+    /* [in] */ __RPC__in_opt IShellItem *psiNewlyCreated);
+    
+    HRESULT STDMETHODCALLTYPE PreNewItem( 
+    /* [in] */ DWORD dwFlags,
+    /* [in] */ __RPC__in_opt IShellItem *psiDestinationFolder,
+    /* [string][unique][in] */ __RPC__in_opt_string LPCWSTR pszNewName);
+    
+    HRESULT STDMETHODCALLTYPE PostNewItem( 
+    /* [in] */ DWORD dwFlags,
+    /* [in] */ __RPC__in_opt IShellItem *psiDestinationFolder,
+    /* [string][unique][in] */ __RPC__in_opt_string LPCWSTR pszNewName,
+    /* [string][unique][in] */ __RPC__in_opt_string LPCWSTR pszTemplateName,
+    /* [in] */ DWORD dwFileAttributes,
+    /* [in] */ HRESULT hrNew,
+    /* [in] */ __RPC__in_opt IShellItem *psiNewItem);
+    
+    HRESULT STDMETHODCALLTYPE UpdateProgress( 
+    /* [in] */ UINT iWorkTotal,
+    /* [in] */ UINT iWorkSoFar);
+    
+    HRESULT STDMETHODCALLTYPE ResetTimer( void);
+    
+    HRESULT STDMETHODCALLTYPE PauseTimer( void);
+    
+    HRESULT STDMETHODCALLTYPE ResumeTimer( void);
+    ULONG m_cRef;
+};
+
+DelFileProgressSink::DelFileProgressSink() {
+  m_cRef = 0;
+}
+DelFileProgressSink::~DelFileProgressSink(void) {
+  m_cRef = 0;
```

Comment:
This should be redundant, the only time any calls delete on this object is you, inside release, and only when m_cRef == 0. I BELIEVE you can make your dtor private, since you are the only one that calls it, that would prevent someone else from accidentally calling it, though since it is the OS using this they are going to do the right thing.